pkg/repo/repo.go

```diff
@@ -45,9 +45,19 @@ func NewFile() *File {
 }
 
 // LoadFile takes a file at the given path and returns a File object
-func LoadFile(path string) (*File, error) {
+// createRepoFile parameter when set to true, will create repo file if
+// it doesn't exist. The parameter is useful for lazy create when
+// running some commands.
+func LoadFile(path string, createRepoFile bool) (*File, error) {
```

Comment:
I was initially of the same thinking but I then switched because I though there might be some scenario where it might want an error is the repo cache does not exist. Is/Could there be such a scenario?

Reply:
well, `helm repo add` is a bit different as it ignores that the files does not exist as it will create it anyway at the end when writing to it if it doesn't exist. I am not sure if this would work for the other `helm repo` commands?

Reply:
if you look at `helm repo list`, it requires a file handle to be returned so it can check for contents (if any repo exists). If none exist then it would return:

```console
Error: no repositories to show
```

Reply:
I would like to add one point to the discussion since it was very confusing for us and seemed unexpected. If you run `helm install cert-manager https://charts.jetstack.io/charts/cert-manager-v0.10.0-alpha.0.tgz` or `helm install cert-manager cert-manager --repo https://charts.jetstack.io --version v0.10.0-alpha.0` these commands will require an existing repository.yaml and cause an error otherwise. It would be great if the `helm install` command could also lazy create the repsitory.yaml.

Reply:
@FabianKramm Can you show the error output? Want to make sure have same issue.

Reply:
@hickeyma Sure, this is the output after PR #6300 is applied:
```
$ rm /Users/fabiankramm/Library/Preferences/helm/repositories.yaml

$ bin/helm install cert-manager cert-manager --repo https://charts.jetstack.io --version v0.10.0-alpha.0
Error: failed to download "https://charts.jetstack.io/charts/cert-manager-v0.10.0-alpha.0.tgz" (hint: running `helm repo update` may help)

$ bin/helm install cert-manager cert-manager --repo https://charts.jetstack.io --version v0.10.0-alpha.0 --debug
install.go:158: [debug] Original chart version: "v0.10.0-alpha.0"
Error: couldn't load repositories file (/Users/fabiankramm/Library/Preferences/helm/repositories.yaml): open /Users/fabiankramm/Library/Preferences/helm/repositories.yaml: no such file or directory
helm.go:81: [debug] open /Users/fabiankramm/Library/Preferences/helm/repositories.yaml: no such file or directory
couldn't load repositories file (/Users/fabiankramm/Library/Preferences/helm/repositories.yaml)
helm.sh/helm/pkg/repo.LoadFile
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/pkg/repo/repo.go:52
helm.sh/helm/pkg/downloader.(*ChartDownloader).ResolveChartVersion
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/pkg/downloader/chart_downloader.go:157
helm.sh/helm/pkg/downloader.(*ChartDownloader).DownloadTo
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/pkg/downloader/chart_downloader.go:87
helm.sh/helm/pkg/action.(*ChartPathOptions).LocateChart
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/pkg/action/install.go:612
main.runInstall
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/cmd/helm/install.go:170
main.newInstallCmd.func1
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/cmd/helm/install.go:109
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).execute
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:762
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:850
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).Execute
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:800
main.main
        /Users/fabiankramm/Programmieren/go-workspace/src/helm.sh/helm/cmd/helm/helm.go:80
runtime.main
        /usr/local/go/src/runtime/proc.go:200
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1337
```

Reply:
So, I have removed the option to create repo. Instead now it will create the repo file if it does not exist. The issues when doing an install direct from the repo, changed my mind! :)

Reply:
@FabianKramm Can you test if this fixes the install direct from remote repo?

Reply:
@hickeyma Yes it works now! Thanks for the fix!

```
$ rm /Users/fabiankramm/Library/Preferences/helm/repositories.yaml

$ bin/helm install cert-manager cert-manager --repo https://charts.jetstack.io --version v0.10.0-alpha.0 -n test
NAME: cert-manager
LAST DEPLOYED: 2019-08-28 16:30:24.900687 +0200 CEST m=+0.579326976
NAMESPACE: test
STATUS: deployed
NOTES:
cert-manager has been deployed successfully!

In order to begin issuing certificates, you will need to set up a ClusterIssuer
or Issuer resource (for example, by creating a 'letsencrypt-staging' issuer).

More information on the different types of issuers and how to configure them
can be found in our documentation:

https://docs.cert-manager.io/en/latest/reference/issuers.html

For information on how to configure cert-manager to automatically provision
Certificates for Ingress resources, take a look at the `ingress-shim`
documentation:

https://docs.cert-manager.io/en/latest/reference/ingress-shim.html
```